lib/Client/Client.php

```diff
@@ -27,14 +28,21 @@ class Client
     private $config;
 
     /**
-     * @param array $options
+     * @var ClientInterface
      */
-    public function __construct(array $options)
+    private $httpClient;
+
+    /**
+     * @param array           $options
+     * @param ClientInterface $httpClient
+     */
+    public function __construct(array $options, ClientInterface $httpClient)
```

Comment:
BC break: it's ok to add the http client injection, but i think it should be optional.
If not injected, a factory can be called to create an http client based on existing classes/extensions and throw an exception if no client can be found. This will allow to fall back to plain `curl_` (or `file_get_contents`) calls